SoftLayer/managers/dedicated_host.py

```diff
@@ -0,0 +1,363 @@
+                                       % datacenter)
+
+    def get_create_options(self):
+        """Returns valid options for ordering a dedicated host."""
+        package = self._get_package()
+        # Locations
+        locations = []
+        for region in package['regions']:
+            locations.append({
+                'name': region['location']['location']['longName'],
+                'key': region['location']['location']['name'],
+            })
+        # flavors
+        dedicated_host = []
+        for item in package['items']:
+            if item['itemCategory']['categoryCode'] == \
+                    'dedicated_virtual_hosts':
+                dedicated_host.append({
+                    'name': item['description'],
+                    'key': item['keyName'],
+                })
+
+        return {
+            'locations': locations,
+            'dedicated_host': dedicated_host,
+        }
+
+    def _get_price(self, package):
+        """Returns valid price for ordering a dedicated host."""
+
+        for price in package['prices']:
+            if not price.get('locationGroupId'):
+                return price['id']
+
+        raise SoftLayer.SoftLayerError(
+            "Could not find valid price")
+
+    def _get_item(self, package, flavor):
+        """Returns the item for ordering a dedicated host."""
+
+        for item in package['items']:
+            if item['keyName'] == flavor:
+                return item
+
+        raise SoftLayer.SoftLayerError("Could not find valid item for: '%s'"
+                                       % flavor)
+
+    def _get_backend_router(self, locations, item):
+        """Returns valid router options for ordering a dedicated host."""
+        mask = '''
+            id,
+            hostname
+        '''
+        cpu_count = item['capacity']
+
+        for capacity in item['bundleItems']:
+            for category in capacity['categories']:
+                if category['categoryCode'] == 'dedicated_host_ram':
+                    mem_capacity = capacity['capacity']
+                if category['categoryCode'] == 'dedicated_host_disk':
+                    disk_capacity = capacity['capacity']
+
+        if locations is not None:
+            for location in locations:
+                if location['locationId'] is not None:
+                    loc_id = location['locationId']
+                    host = {
+                        'cpuCount': cpu_count,
+                        'memoryCapacity': mem_capacity,
+                        'diskCapacity': disk_capacity,
+                        'datacenter': {
+                            'id': loc_id
+                        }
+                    }
+                    routers = self.host.getAvailableRouters(host, mask=mask)
+                    return routers
+
+        raise SoftLayer.SoftLayerError("Could not find available routers")
+
+    def _get_default_router(self, routers, router_name):
```

Comment:
I think for readability, since router_name isn't required, default it to None in the args here. Even though you always supply it above (and it's None because it's optional in _generate_create_dict()), I'd still be explicit with `router_name=None`